addons/website_payment/models/payment_provider.py

```diff
@@ -25,6 +25,11 @@ def _get_compatible_providers(self, *args, website_id=None, **kwargs):
         :rtype: payment.provider
         """
         providers = super()._get_compatible_providers(*args, website_id=website_id, **kwargs)
+        # retrieve website_id from the order if it is defined for the correct portal view
+        if not website_id and kwargs.get('sale_order_id'):
+            order = request.env['sale.order'].sudo().browse(kwargs.get('sale_order_id')).exists()
```

Comment:
When sudoing, it's better to indicated it in the variable name, here `order_sudo` or `order_su`. This way, it's easier to remember the order is in sudo, and we need to be careful to what we are doing with it.